app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class ReservationsController < ApplicationController
+  include JSONAPI::Fetching
+
+  def index
+    reservations = policy_scope(Reservation).where(screening: params[:screening_id])
+
+    render jsonapi: reservations, except: blacklisted_attributes
+  end
+
+  def show
+    reservation = policy_scope(Reservation).find(params[:id])
```

Comment:
Ok, for show it would be better to do
```ruby
reservation = Resevation.find(params[:id])
authorize reservation
```
That way, when accessing a reservation that I do not have access to, it will return an unauthorized error from pundit and that will be translated to correct unauthorized message in your API. Your current code would return a 404 error instead.